### PR TITLE
Drivers & ADRV9009 project updates.

### PR DESCRIPTION
**Pull request by amiclaus**
1. Update drivers using `axi_io` wrapper functions. Make drivers platform agnostic.

2. Update ADRV9009 main function using `axi_io` functions and platform specific initialization parameters.

Signed-off-by: Antoniu Miclaus <antoniu.miclaus@analog.com>

### CODE REVIEW

**Comment by amiclaus** [comment]
Changelog v1->v2: 

- fix code style issues

File: projects/drivers/axi_adc_core/axi_adc_core.h [hunk 1]
```diff
@@ -52,12 +52,16 @@ struct axi_adc {
 	uint32_t base;
 	uint8_t	num_channels;
 	uint64_t clock_hz;
+	int32_t (*axi_io_read)(uint32_t, uint32_t, uint32_t*);
```
Comment (mirceacaprioru) [inline projects/drivers/axi_adc_core/axi_adc_core.h]: why does ad9528 has the extra parameter and here we have the read/write function prototype ?

Comment (amiclaus) [reply projects/drivers/axi_adc_core/axi_adc_core.h]: The `extra` void pointer is used to point at platform specific parameters (i.e. Xilinx/Altera spi config params) while `axi_io_read` and `axi_io_write` are pointers to functions that wrap read/write platform specific functions for AXI IO interface.

File: projects/drivers/axi_dac_core/axi_dac_core.c [hunk 2]
```diff
@@ -796,16 +772,9 @@ int32_t axi_dac_set_buff(struct axi_dac *dac,
...
 		data_q = (buff[index + 1] << 16);
-#ifdef ALTERA_PLATFORM
-		IOWR_32DIRECT(address, index * 2, data_i | data_q);
-#else
-		Xil_Out32(address + index * 2, data_i | data_q);
-#endif
-	}
 
-#ifndef ALTERA_PLATFORM
-	Xil_DCacheFlush();
```
Comment (adrimbarean) [inline projects/drivers/axi_dac_core/axi_dac_core.c]: This functionality is completely removed. Don't we still need this?

File: projects/drivers/jesd204/axi_jesd204_rx.h [hunk 3]
```diff
@@ -73,6 +75,8 @@ struct jesd204_rx_init {
 	uint8_t subclass;
 	uint32_t device_clk_khz;
 	uint32_t lane_clk_khz;
+	int32_t (*axi_io_read)(uint32_t, uint32_t, uint32_t*);
+	int32_t(*axi_io_write)(uint32_t, uint32_t, uint32_t);
```
Comment (adrimbarean) [inline projects/drivers/jesd204/axi_jesd204_rx.h]: Missing space here. Does it still compile like this?

File: projects/drivers/axi_dac_core/axi_dac_core.c (hunk 2 above)
Comment (amiclaus) [reply projects/drivers/axi_dac_core/axi_dac_core.c]: The project is working without the Cache flush. Keeping these functionalities will imply adding a special wrapper for this function to make the driver platform agnostic (besides axi_io read/write). So, I believe this subject is debatable.

Comment (dbogdan) [reply projects/drivers/axi_dac_core/axi_dac_core.c]: We need it - it should be moved to the main() function.

**Comment by amiclaus** [comment]
Changelog v2->v3:

- add missing space
- perform cache flush inside main function after loading custom data.

**Pull request merged**